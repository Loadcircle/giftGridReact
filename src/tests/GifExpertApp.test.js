const { shallow } = require("enzyme")
const { GifExpertApp } = require("../GifExpertApp")
import React from 'react';
import '@testing-library/jest-dom';

describe('Pruebas en gif expert app', ()=>{

    const wrapper = shallow(<GifExpertApp/>);

    test('Snapshot match', ()=>{

        expect(wrapper).toMatchSnapshot();

    })

    test('Debe de mostrar una lista de categoris', ()=>{

        const categories = ['One Punch', 'DBZ']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toMatchSnapshot(categories.length);

    });

})