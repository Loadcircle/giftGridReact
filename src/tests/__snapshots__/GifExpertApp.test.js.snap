// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en gif expert app Debe de mostrar una lista de categoris 1`] = `
<div>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="DBZ"
      key="DBZ"
    />
  </ol>
</div>
`;

exports[`Pruebas en gif expert app Debe de mostrar una lista de categoris 2`] = `2`;

exports[`Pruebas en gif expert app Snapshot match 1`] = `
<div>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
